pkg/stream/client.go

```diff
@@ -799,3 +799,26 @@ func (c *Client) DeclareSubscriber(streamName string,
 	}()
 	return consumer, err.Err
 }
+
+func (c *Client) StreamStats(streamName string) (*StreamStats, error) {
+
+	resp := c.coordinator.NewResponse(commandStreamStatus)
+	correlationId := resp.correlationid
+
+	length := 2 + 2 + 4 + 2 + len(streamName)
+
+	var b = bytes.NewBuffer(make([]byte, 0, length+4))
+	writeProtocolHeader(b, length, commandStreamStatus,
+		correlationId)
+	writeString(b, streamName)
+
+	err := c.handleWriteWithResponse(b.Bytes(), resp, false)
+	offset := <-resp.data
+
+	_ = c.coordinator.RemoveResponseById(resp.correlationid)
+	if err.Err != nil {
+		return nil, err.Err
+	}
+	m := offset.(map[string]int64)
```

Comment:
This is unsafe as it can panic if the type assertion is not successful. It's probably a bug if it panics, but perhaps we could return an error asking to report a bug if the assignment is not successful in the safer `v, ok := T.(x)` [version](https://go.dev/ref/spec#Type_assertions)?

Reply:
>Which require tribal knowledge of the protocol type sizes and structure of the frames. 

you are right, but atm all the client is done in this way. Let's do that in `v2`